fun/middleware/enforce_legal_acceptance.py

```diff
@@ -3,72 +3,61 @@
-        r"""^/(news|login|contact|about|help|legal|privacy|honor|charte)/?$""",
-        r""".*/login_ajax.*""",
-        r"""^/static.*""",
-        r""".*search.*""",
-        r""".*revision.*""",
-        r"""^/static/.*""",
-        r"""^/api/user/.*""",
-        r"""^/c4x.*""",
-        r""".*logout.*$""",
-        r""".*asset.*""",
-        r""".*\.(png|font|html|js|css|jpeg|pdf|gif|mp4).*""",
-        r"""^/?$""",
+        r'^/payment/.*',
+        r'^/xblock/.*',
+        r'^/account/setting.*',
+        r'.*accept.*',
+        r'^/(news|login|contact|about|help|legal|privacy|honor|charte)/?$',
+        r'.*/login_ajax.*',
+        r'^/static.*',
+        r'.*search.*',
+        r'.*revision.*',
+        r'^/static/.*',
+        r'^/api/user/.*',
+        r'^/c4x.*',
+        r'.*logout.*$',
+        r'.*asset.*',
+        r'.*\.(png|font|html|js|css|jpeg|pdf|gif|mp4).*',
+        r'^/?$',
     )
 )
-def get(self, key, default):
-    return getattr(self, key) if hasattr(self, key) else default
 
-LMS_BASE = get(settings, "LMS_BASE", "")
-PROT = get(settings, "HTTPS", "off") == "on" and "s" or ""
-LMS_REDIR = LMS_BASE and ("http%s://%s/" % (PROT, LMS_BASE)) or ""
-DEFAULT_AGREEMENT_FORM = "%s/payment/terms/" % (LMS_REDIR)
+TERMS_AND_CONDITIONS_AGREEMENT = '/payment/terms/'
+
 
 def terms_accepted(func):
     """
     force redirect to acceptance of legal condition if
     latest not accepted
-    redirection landing page SHOULD be set in settings
-    AGREEMENT_FORM else defaults to DEFAULT_AGREEMENT_FORM
     """
     def wrapped(request, *args, **kwargs):
-
         if not legal_acceptance(request.user):
-            return redirect(
-                get(settings, "AGREEMENT_FORM", DEFAULT_AGREEMENT_FORM),
-                *args, **kwargs
-            )
+            return redirect(TERMS_AND_CONDITIONS_AGREEMENT, *args, **kwargs)
         else:
             return func(request, *args, **kwargs)
     return wrapped
 
+
 class LegalAcceptance(object):
-    def __init__(self, *a, **kw):
-        self.white_list = get(
-            settings,
-            "AGREEMENT_WHITELIST",
-            DEFAULT_AGREEMENT_WHITELIST
-        )
 
     def is_whitelisted(self, url):
-        return any(wl_url.match(url) for wl_url in self.white_list)
+        return any(wl_url.match(url) for wl_url in AGREEMENT_WHITELIST)
 
     def process_view(self, request, view_func, view_args, view_kwargs):
-        if not self.is_whitelisted(request.path_info) \
-            and hasattr(request, "user") \
-            and request.user.is_authenticated and \
-            not legal_acceptance(request.user):
+        if (
+                not self.is_whitelisted(request.path_info) and \
+                hasattr(request, "user") and \
+                request.user.is_authenticated() and \
+                # don't enforce legal acceptance when impersonating a user
+                not request.user.is_masked and \
+                not legal_acceptance(request.user)):
```

Comment:
I'm not a big fan of this indentation style. What about the following with a comment on top of the block detailing each condition?

```python
# We redirect current request user to the terms acceptance view if:
#   1. the current url is whitelisted
#   2. ...
if not self.is_whitelisted(request.path_info) and \
        hasattr(request, "user") and \
        request.user.is_authenticated() and \
        not request.user.is_masked and \
        not legal_acceptance(request.user):
```

Reply:
and w can remove the "\" when we use the parenthese.

Reply:
You are right: not aligning conditions could decrease readability. In this case, here is a new proposal:

```python
# We redirect current request user to the terms acceptance view if:
#   1. the current url is whitelisted
#   2. ...
if (not self.is_whitelisted(request.path_info) and \
    hasattr(request, "user") and \
    request.user.is_authenticated() and \
    not request.user.is_masked and \
    not legal_acceptance(request.user)):
```

WDYT?

Reply:
(comment updated to remove unnecessary back slashes)